utility.hpp.in

```diff
@@ -13,17 +13,28 @@
  */
 namespace utility{
 
-/* forward Declaration of function signatures */
+/* functions */
+/** @returns A copy of the argument value */
+template< typename T >
+T   copy( const T& t ){ return t; }
```

Comment:
What I meant was your last example:
```cpp
void foo( expensiveType e);
```
and
```cpp
expansiveType bar;
foo( std::move(copy) 
```
It is a good thing to have these discussions. I like your approach. We actually know we are copying/moving expensive types. Both approaches are essentially equivalent but differ in the sense that one will copy by default while the other will move by default unless we specifically say otherwise. And if we have to std::move all the time, we are probably doing it wrong.